plugins/user.py

```diff
@@ -0,0 +1,454 @@
+from will.plugin import WillPlugin
+from will.decorators import respond_to, require_settings
+from will import settings
+import pyotp
+import qrcode
+import base64
+import boto
+import boto.s3.connection
+from boto.s3.key import Key
+import cStringIO
+import datetime
+import urlparse
+from simplecrypt import encrypt, decrypt
+import pytz
+
+import logging
+log = logging.getLogger(__name__)
+
+
+# TODO: QR code over hipchat leaves the account vulnerable if the client renders the image and leaves it rendered in the chat history
```

Comment:
See other comment - I think we're safe from replays as long as the person tabs out of the alton 1-on-1 chat at some point.